go/vt/vtgate/plugin_mysql_server_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreedto in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vtgate
+
+import (
+	"golang.org/x/net/context"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/youtube/vitess/go/mysql"
+	"github.com/youtube/vitess/go/sqltypes"
+)
+
+type testHandler struct {
+	lastConn *mysql.Conn
+}
+
+func (th *testHandler) NewConnection(c *mysql.Conn) {
+	th.lastConn = c
+}
+
+func (th *testHandler) ConnectionClosed(c *mysql.Conn) {
+}
+
+func (th *testHandler) ComQuery(c *mysql.Conn, q []byte, callback func(*sqltypes.Result) error) error {
+	return nil
+}
+
+func TestConnectionUnixSocket(t *testing.T) {
+	th := &testHandler{}
+
+	authServer := mysql.NewAuthServerStatic()
+
+	authServer.Entries["user1"] = []*mysql.AuthServerStaticEntry{
+		{
+			Password:   "password1",
+			UserData:   "userData1",
+			SourceHost: "localhost",
+		},
+	}
+
+	unixSocket := "/tmp/mysql_vitess_test.sock"
```

Comment:
The hardcoded path means that two different users/builds cannot run this test at the same time.

How about reserving a temp file via `ioutil.TempFile` and then use its path instead?

https://golang.org/pkg/io/ioutil/#example_TempFile